src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1492,25 +1516,18 @@ OptoReg::Name PhaseChaitin::bias_color(LRG& lrg) {
     }
   }
 
+  // Try biasing the color with non-interfering first input's lrg.
   uint copy_lrg = _lrg_map.find(lrg._copy_bias);
-  if (copy_lrg != 0) {
-    // If he has a color,
-    if(!_ifg->_yanked->test(copy_lrg)) {
-      OptoReg::Name reg = lrgs(copy_lrg).reg();
-      //  And it is legal for you,
-      if (is_legal_reg(lrg, reg)) {
-        return reg;
-      }
-    } else if (!lrg.mask().is_offset()) {
-      // Choose a color which is legal for him
-      ResourceMark rm(C->regmask_arena());
-      RegMask tempmask(lrg.mask(), C->regmask_arena());
-      tempmask.and_with(lrgs(copy_lrg).mask());
-      tempmask.clear_to_sets(lrg.num_regs());
-      OptoReg::Name reg = find_first_set(lrg, tempmask);
-      if (OptoReg::is_valid(reg))
-        return reg;
-    }
+  OptoReg::Name reg = select_bias_lrg_color(lrg, copy_lrg);
+  if (reg != OptoReg::Bad) {
```

Comment:
Please, use `OptoReg::is_valid(reg)` here. I find it more readable.

Also, there's repetitive pattern for `lrg._copy_bias` and `lrg._copy_bias2`. Would be nice to hide it behind a single `select_bias_lrg_color(_lrg_map, lrg)` call.

Reply:
Done.